lib/grandpa/message_handler_test.go

```diff
@@ -587,3 +587,248 @@ func TestMessageHandler_VerifyBlockJustification(t *testing.T) {
 	err = gs.VerifyBlockJustification(testHash, data)
 	require.Equal(t, ErrMinVotesNotMet, err)
 }
+
+func Test_getEquivocatoryVoters(t *testing.T) {
+	// many of equivocatory votes
+	ed25519Keyring, _ := keystore.NewEd25519Keyring()
+	fakeAuthorities := []*ed25519.Keypair{
+		ed25519Keyring.Alice().(*ed25519.Keypair),
+		ed25519Keyring.Alice().(*ed25519.Keypair),
+		ed25519Keyring.Alice().(*ed25519.Keypair),
+		ed25519Keyring.Bob().(*ed25519.Keypair),
+		ed25519Keyring.Charlie().(*ed25519.Keypair),
+		ed25519Keyring.Charlie().(*ed25519.Keypair),
+		ed25519Keyring.Charlie().(*ed25519.Keypair),
+		ed25519Keyring.Dave().(*ed25519.Keypair),
+		ed25519Keyring.Dave().(*ed25519.Keypair),
+		ed25519Keyring.Eve().(*ed25519.Keypair),
+		ed25519Keyring.Ferdie().(*ed25519.Keypair),
+		ed25519Keyring.Heather().(*ed25519.Keypair),
+		ed25519Keyring.Heather().(*ed25519.Keypair),
+		ed25519Keyring.Ian().(*ed25519.Keypair),
+		ed25519Keyring.Ian().(*ed25519.Keypair),
+	}
+
+	authData := make([]AuthData, len(fakeAuthorities))
+
+	for i, auth := range fakeAuthorities {
+		authData[i] = AuthData{
+			AuthorityID: auth.Public().(*ed25519.PublicKey).AsBytes(),
+		}
+	}
+
+	eqv := getEquivocatoryVoters(authData)
+	require.Len(t, eqv, 5)
+}
+
+func Test_VerifyCommitMessageJustification_ShouldRemoveEquivocatoryVotes(t *testing.T) {
+	const fakeRound = 2
+
+	gs, st := newTestService(t)
+	h := NewMessageHandler(gs, st.Block)
+
+	// add 10 previous blocks and returns the last one
+	bfcBlock := addBlocksAndReturnTheLastOne(t, st.Block, 8, time.Now())
```

Comment:
Isn't it `8` blocks perhaps?

Also maybe change to

```suggestion
	const previousBlocksToAdd = 8
	bfcBlock := addBlocksAndReturnTheLastOne(t, st.Block, previousBlocksToAdd, time.Now())
```

so that the comment is no longer needed.

And final question, could we mock the time here? in example `now := time.Unix(1000, 0)`?

:bulb: this comment applies for tests below (search for `// add 10 previous blocks` comments)

Reply:
Done to all tests with the comment!